shared/lib/deep-links/parse.ts

```
@@ -18,7 +20,11 @@ export async function parse(url: URL): Promise<ParsedDeepLink | false> {
 
   let destination: Destination;
   try {
-    destination = route.handler(url.searchParams);
+    const canonicalUrl = new URL(canonicalize(url));
```

`verify` also does `canonicalize(url)`.

So you _could_ just do:

```typescript
const canonicalUrl = canonicalize(url); // make canonicalize return a `URL` instead of a string
const signatureStr = url.searchParams.get("sig");
const signature = sig ? await verify(signatureStr, canonicalUrl) : MISSING;
try {
    canonicalUrl.delete(SIG_PARAMS);
    destination = route.handler(routeParams);
}
// ...
```

and update `verify` to:

a) require the `sig` in its function signature
b) no longer check if `sig` is present, since we can just do that here
c) no longer use `canonicalize`, as the URL is already `canonicalized`
d) accept a `URL` instead of a string (and then do this: `encoder.encode(canonicalUrl.toString())`)

This changes the order we process values. Previously we wouldn't call `verify` if the route wasn't valid. But in reality, I don't think this matters. I did it this way initially just to "fail fast", but I its not neccessary.